src/lamp_py/ingestion_tm/jobs/parition_table.py

```diff
@@ -182,8 +160,122 @@ def run_export(self, tm_db: MSSQLManager) -> None:
                     )
                     upload_file(local_pq, s3_export_path)
 
-            self.update_version_file()
-            logger.log_complete()
+                logger.log_complete()
 
-        except Exception as exception:
-            logger.log_failure(exception)
+            except Exception as exception:
+                logger.log_failure(exception)
+
+        self.update_version_file()
+
+
+class TMDailyLogStopCrossing(TMDailyTable):
+    """Export STOP_CROSSING table from TMDailyLog"""
+
+    def __init__(self) -> None:
+        TMDailyTable.__init__(
+            self,
+            s3_location=RemoteFileLocations.tm_stop_crossing,
+            tm_table="TMDailyLog.dbo.STOP_CROSSING",
+            lamp_version="0.0.1",
+        )
+
+    @property
+    def export_schema(self) -> pyarrow.schema:
+        return pyarrow.schema(
+            [
+                ("STOP_CROSSING_ID", pyarrow.int64()),
+                ("TRIP_GEO_NODE_XREF_ID", pyarrow.int64()),
+                ("PATTERN_GEO_NODE_SEQ", pyarrow.int64()),
+                ("CALENDAR_ID", pyarrow.int64()),
+                ("ROUTE_DIRECTION_ID", pyarrow.int64()),
+                ("PATTERN_ID", pyarrow.int64()),
+                ("GEO_NODE_ID", pyarrow.int64()),
+                ("BLOCK_STOP_ORDER", pyarrow.int64()),
+                ("SCHEDULED_TIME", pyarrow.int64()),
+                ("ACT_ARRIVAL_TIME", pyarrow.int64()),
+                ("ACT_DEPARTURE_TIME", pyarrow.int64()),
+                ("ODOMETER", pyarrow.int64()),
+                ("WAIVER_ID", pyarrow.int64()),
+                ("DAILY_WORK_PIECE_ID", pyarrow.int64()),
+                ("TIME_POINT_ID", pyarrow.int64()),
+                ("SERVICE_TYPE_ID", pyarrow.int64()),
+                ("VEHICLE_ID", pyarrow.int64()),
+                ("TRIP_ID", pyarrow.int64()),
+                ("PULLOUT_ID", pyarrow.int64()),
+                ("IsRevenue", pyarrow.string()),
+                ("SCHEDULE_TIME_OFFSET", pyarrow.int64()),
+                ("CROSSING_TYPE_ID", pyarrow.int64()),
+                ("OPERATOR_ID", pyarrow.int64()),
+                ("CANCELLED_FLAG", pyarrow.bool_()),
+                ("IS_LAYOVER", pyarrow.bool_()),
+                ("ROUTE_ID", pyarrow.int64()),
+            ]
+        )
+
+
+class TMDailyLogDailyWorkPiece(TMDailyTable):
```

Comment:
This table links operators to pieces of work, allowing us to figure out who drove what runs, blocks, and trips.